src/parallelization/hpxdataflowsimulator.h

```diff
@@ -109,19 +112,22 @@ class CellComponent
         std::vector<hpx::future<void> > remoteIDFutures;
         remoteIDFutures.reserve(neighbors.size());
 
+	hpx::lcos::local::spinlock mutex;
```

Comment:
hmm, above you pull in the standard mutex header, but here you instantiate a spinlock from HPX. Is the include wrong or this type or am I missing something?